config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,210 @@
+              curl --silent --max-time 5 http://localhost:8080 &
+              p2=$!
+
+              # Check GPU usage (If worker node restarted, it becomes NOT ready)
+              gpu_status=$(ray status | grep GPU)
+              used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Readiness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi & # Run this check in the background
+              p3=$!
+
+              # Wait for all processes to complete
+              wait $p1 $p2 $p3
+              wait $p1 && wait $p2 && wait $p3
+      startupProbe:
+        failureThreshold: 40
+        periodSeconds: 30
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              ray status
+  volumes:
+    - name: shm
+      emptyDir:
+        medium: Memory
+        sizeLimit: 12Gi
+  workerSpec:
+    containers:
+      - name: worker-container
+        image: kserve/huggingfaceserver:latest
+        command: ["bash", "-c"]
+        args:
+          - |
+            SECONDS=0
+
+            while true; do              
+              if (( SECONDS <= 120 )); then
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379" > /dev/null 2>&1; then
+                  echo "GCS is ready."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Waiting for GCS to be ready."
+              else
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"; then
+                  echo "GCS is ready. Any error messages above can be safely ignored."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Still waiting for GCS to be ready."
+                echo "For troubleshooting, refer to the FAQ at https://github.com/ray-project/kuberay/blob/master/docs/guidance/FAQ.md."
+              fi
+              
+              sleep 5
+            done
+
+            echo "Attempting to connect to Ray cluster at $RAY_HEAD_ADDRESS ..."
+            RAY_HEAD_ADDRESS="${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"
+            ray start --address="$RAY_HEAD_ADDRESS" --block
+        env:
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace              
+        resources:
+          limits:
+            cpu: "16"
+            memory: 48Gi
```

Comment:
wow, isn't it too much?

Reply:
If users choose to use multi-node/multi-gpu, they would try to load a big model. Then memory and CPU should be big enough to handle model loading, data preprocessing/postprocessing, communication, and management tasks. So from my point of view, this is not too big. (FYI, this resource is based on 70B model)

Reply:
can we add this info on the docs you wrote?

Reply:
I can understand the motivation of the large amount of memory, but I do not understand the large limit, nor the large request of CPUs, as models would use mainly GPUs, rather than CPUs. And this is per worker, so I'm seeing it as too much CPU per worker.

Perhaps, let's use a more conservative number in `requests`, while giving enough room in `limits`.

Reply:
what do you recommend?